### PR TITLE
test: add spreadsheet grouping unit tests

### PR DESCRIPTION
**Pull request by tomivirkki**
Add Junit tests for the row/column grouping feature, especially targeting the [GroupingUtil](https://github.com/vaadin/flow-components/blob/master/vaadin-spreadsheet-flow-parent/vaadin-spreadsheet-flow/src/main/java/com/vaadin/flow/component/spreadsheet/GroupingUtil.java) class

### CODE REVIEW

File: vaadin-spreadsheet-flow-parent/vaadin-spreadsheet-flow/src/test/java/com/vaadin/flow/component/spreadsheet/tests/GroupingTest.java [hunk 1]
```diff
@@ -0,0 +1,327 @@
...
+        collapseRow(SHEET4_PARENT_ROW_GROUP);
+        Assert.assertTrue(spreadsheet.isRowHidden(SHEET4_ROW));
+
+        // Expand Parent row group
+        expandRow(SHEET4_PARENT_ROW_GROUP);
+        Assert.assertFalse(spreadsheet.isRowHidden(SHEET4_ROW));
+    }
+
+    @Test
+    public void inverted_collapseRow_rowHidden() {
```
Comment (DiegoCardoso) [inline vaadin-spreadsheet-flow-parent/vaadin-spreadsheet-flow/src/test/java/com/vaadin/flow/component/spreadsheet/tests/GroupingTest.java:86]: Just a clarification: what does inverted here refer to?

**Review by DiegoCardoso — APPROVED** [review]
LGTM. Just one clarification question.

File: vaadin-spreadsheet-flow-parent/vaadin-spreadsheet-flow/src/test/java/com/vaadin/flow/component/spreadsheet/tests/GroupingTest.java (hunk 1 above)
Comment (tomivirkki) [reply vaadin-spreadsheet-flow-parent/vaadin-spreadsheet-flow/src/test/java/com/vaadin/flow/component/spreadsheet/tests/GroupingTest.java:86]: Inverted groups is a feature in spreadsheet

regular groups:
Screenshot 2022-10-14 at 17 16 48

inverted groups:
Screenshot 2022-10-14 at 17 16 53

Comment (tomivirkki) [reply vaadin-spreadsheet-flow-parent/vaadin-spreadsheet-flow/src/test/java/com/vaadin/flow/component/spreadsheet/tests/GroupingTest.java:86]: Well, probably doesn't go by that name officially, but that naming was used in the GroupingUtil source code

**Pull request merged**